google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java

```diff
@@ -123,34 +122,43 @@ void waitForExpectedMessages() throws InterruptedException {
       SettableFuture<AckReply> reply = SettableFuture.create();
 
       if (explicitAckReplies) {
-        outstandingMessageReplies.add(reply);
-      } else {
-        if (error.isPresent()) {
-          reply.setException(error.get());
-        } else {
-          reply.set(ackReply);
+        try {
+          outstandingMessageReplies.put(reply);
+        } catch (InterruptedException e) {
+          throw new IllegalStateException(e);
         }
+      } else {
+        replyTo(reply);
       }
 
       return reply;
     }
 
     public void replyNextOutstandingMessage() {
       Preconditions.checkState(explicitAckReplies);
-
-      SettableFuture<AckReply> reply = outstandingMessageReplies.poll();
-      if (error.isPresent()) {
-        reply.setException(error.get());
-      } else {
-        reply.set(ackReply);
+      try {
+        replyTo(outstandingMessageReplies.take());
+      } catch (InterruptedException e) {
+        throw new IllegalStateException(e);
       }
     }
 
     public void replyAllOutstandingMessage() {
       Preconditions.checkState(explicitAckReplies);
+      for (; ; ) {
+        SettableFuture<AckReply> reply = outstandingMessageReplies.poll();
+        if (reply == null) {
+          return;
+        }
+        replyTo(reply);
+      }
```

Comment:
nit: up to you but the following code looks more readable to me

SettableFuture<AckReply> reply;
while ((reply = test.poll()) != null) {
  replyTo(reply);
}